### PR TITLE
Xin 163

### PR DESCRIPTION
**Pull request by wjrjerome**

### CODE REVIEW

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 1]
```diff
@@ -1054,3 +1055,14 @@ func (x *XDPoS_v2) allowedToSend(chain consensus.ChainReader, blockHeader *types
...
 	return nil
 }
+
+// Periodlly execution(Attached to engine initialisation during "new"). Used for pool cleaning etc
+func (x *XDPoS_v2) periodicJob() {
+	go func() {
+		for {
+			<-time.After(utils.PeriodicJobPeriod * time.Second)
+			x.hygieneVotePool()
+			x.hygieneTimeoutPool()
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go:1053]: Need lock here?

File: consensus/XDPoS/utils/types.go [hunk 2]
```diff
@@ -168,7 +172,7 @@ func TimeoutSigHash(m *TimeoutForSign) common.Hash {
 
 func (m *Vote) PoolKey() string {
 	// return the voted block hash
-	return m.ProposedBlockInfo.Hash.Hex()
+	return fmt.Sprint(m.ProposedBlockInfo.Round, ":", m.ProposedBlockInfo.Number, ":", m.ProposedBlockInfo.Hash.Hex())
```
Comment (wgr523) [inline consensus/XDPoS/utils/types.go]: Please also add m.GapNumber in my recent PR

Comment (liam-lai) [reply consensus/XDPoS/utils/types.go]: I think we need to discuss the pool key a bit.

File: consensus/XDPoS/engines/engine_v2/engine.go (hunk 1 above)
Comment (wjrjerome) [reply consensus/XDPoS/engines/engine_v2/engine.go:1053]: Why? they are cleaning up old rounds, it shouldn't need any locks as it's not an important operation.

File: consensus/XDPoS/utils/types.go (hunk 2 above)
Comment (liam-lai) [reply consensus/XDPoS/utils/types.go]: 👍

**Review by liam-lai — APPROVED** [review]

**Pull request merged**